experimental/Schemes/singular_locus.jl

```diff
@@ -52,25 +75,53 @@ function _singular_locus_with_decomposition(X::AbsSpec{<:Ring, <:MPolyQuoLocaliz
   return result
 end
 
-@attr Bool function is_equidimensional(X::AbsSpec{<:Ring, <:MPolyQuoLocalizedRing})
+### only for users' convenience: appropriate return value for NonQuoRings
+
+function singular_locus(X::AbsSpec{<:Ring, <:NonQuoRing})
+  return subscheme(X,ideal(OO(X),one(OO(X))))
+end
+
+function singular_locus_reduced(X::AbsSpec{<:Ring, <:NonQuoRing})
+  return subscheme(X,ideal(OO(X),one(OO(X))))
+end
+
+### to keep number of cases down, create a dummy for 
+### saturated_ideal for the affine case
+function saturated_ideal(I::MPolyIdeal)
+  return(I)
+end
+
+@attr Bool function is_equidimensional(X::AbsSpec{<:Ring, <:QuoRing})
   I = modulus(OO(X))
-  P = equidimensional_decomposition_weak(saturated_ideal(I))
+  P = equidimensional_decomposition_radical(saturated_ideal(I))
   length(P) < 2 && return true
   return false
 end
 
-@attr function reduced_scheme(X::AbsSpec{<:Ring, <:MPolyQuoLocalizedRing})
+@attr function reduced_scheme(X::AbsSpec{<:Ring, <:QuoRing})
   I = modulus(OO(X))
   J = radical(saturated_ideal(I))
```

Comment:
```suggestion
  J = radical(Ocar.pre_saturated_ideal(I))
```
This should avoid one primary decomposition.